client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-location-list.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import ExtendedHeader from 'woocommerce/components/extended-header';
+import List from 'woocommerce/components/list/list';
+import ListItem from 'woocommerce/components/list/list-item';
+import ListHeader from 'woocommerce/components/list/list-header';
+import ListItemField from 'woocommerce/components/list/list-item-field';
+import LocationFlag from 'woocommerce/components/location-flag';
+import ShippingZoneLocationDialog from './shipping-zone-location-dialog';
+import { decodeEntities } from 'lib/formatting';
+import { bindActionCreatorsWithSiteId } from 'woocommerce/lib/redux-utils';
+import { getCurrentlyEditingShippingZoneLocationsList } from 'woocommerce/state/ui/shipping/zones/locations/selectors';
+
+const ShippingZoneLocationList = ( { siteId, loaded, translate, locations, onChange } ) => {
+	const getLocationDescription = ( location ) => {
+		switch ( location.type ) {
+			case 'continent':
+				return translate( 'All countries' );
+			case 'country':
+				if ( location.postcodeFilter ) {
+					return translate( 'Specific postcodes' );
```

Comment:
Since there is only one postcode allowed, maybe we could just print it here? `postcodeFilter` is a string.